comms/src/inbound_message_service/msg_processing_worker.rs

```diff
@@ -141,70 +140,27 @@ mod test {
 
     #[test]
     fn test_new_and_start() {
-        // Create a dispatchable message type with feedback to test the functionality of the processing worker
-        #[derive(PartialEq, Debug)]
-        pub enum DispatchType {
-            Unused,
-            Type1,
-            Type2,
-        }
-
-        #[derive(Debug)]
-        pub struct Message {
-            pub data: String,
-        }
-        impl Message {
-            pub fn to_frame_set(&self) -> Result<FrameSet, MessageError> {
-                let mut frame_set: Vec<Vec<u8>> = Vec::new();
-                frame_set.push(self.data.as_bytes().to_vec());
-                Ok(frame_set)
-            }
-        }
-
-        impl Dispatchable for Message {
-            fn dispatch_type(&self) -> u32 {
-                match self.data.as_ref() {
-                    "Type1" => DispatchType::Type1 as u32,
-                    _ => DispatchType::Type2 as u32,
-                }
-            }
-        }
-        impl TryFrom<FrameSet> for Message {
-            type Error = MessageError;
-
-            fn try_from(frames: FrameSet) -> Result<Self, Self::Error> {
-                if frames.len() == 1 {
-                    Ok(Message {
-                        data: String::from_utf8(frames[0].clone()).unwrap(),
-                    })
-                } else {
-                    Err(MessageError::MalformedMultipart)
-                }
-            }
-        }
-
         // Create a common variable that the worker can change and the test can read to determine that the message was
         // correctly dispatched
-        static mut CALLED_FN_TYPE: DispatchType = DispatchType::Unused;
+        static mut CALLED_FN_TYPE: CommsDispatchType = CommsDispatchType::Discard;
 
         // Setup a test message dispatcher
-        fn test_fn1(_msg_data: Message) -> Result<(), DispatchError> {
-            unsafe {
-                CALLED_FN_TYPE = DispatchType::Type1;
-            }
-            Ok(())
-        }
-
-        fn test_fn2(_msg_data: Message) -> Result<(), DispatchError> {
+        //        fn test_fn_forward(_msg_data: MessageContext<RistrettoPublicKey>) -> Result<(), DispatchError> {
+        //            unsafe {
+        //                CALLED_FN_TYPE = CommsDispatchType::Forward;
+        //            }
+        //            Ok(())
+        //        }
+
+        fn test_fn_handle(_msg_data: MessageContext<RistrettoPublicKey>) -> Result<(), DispatchError> {
```

Comment:
This test no longer tests the dispatching to multiple handlers.

Reply:
That is correct - InboundMessageServiceResolver needs to be implemented with tests. There's work to be done here.